common-lib/src/cmr/common/util.clj

```diff
@@ -163,11 +163,48 @@
          result# (do ~@body)]
      [(- (System/currentTimeMillis) start#) result#]))
 
+(defmacro defn-timed-level
+  "Creates a function that logs how long it took to execute the body. It
+   supports multiarity functions but only times how long the last listed arity
+   version takes. This means it should be used with multiarity functions where
+   it calls itself with the extra arguments.
+   This function is different from defn-timed by allowing the log function to be
+   passed in so as to allow the calling function to pick which level to log to."
+  [fn-name level & fn-tail]
+  (let [fn-name-str (name fn-name)
```

Comment:
Can you first check the logging level before doing all of this work and if the level is below the current setting then don't do the code.?

Reply:
Or is the level being passed in what to set this particular log level to - which then my previous comment doesn't make sense.

Reply:
the function still has to be executed, so I can't skip all of it, but I was thinking of adding a defconf to allow the level to be changed.